evals/cybench/dataset.py

```diff
@@ -0,0 +1,150 @@
+        target = challenge_info["flag"]
+        challenge_files = _get_files(challenge_info.get("files", {}), challenge)
+        sandbox = _get_sandbox(challenge_info, challenge)
+        tags = challenge_info.get("tags", [])
+        variants = challenge_info["variants"]
+
+        samples = []
+        if isinstance(variants, list):
+            variants = {
+                f"variant_{i}": {"prompt": variant}
+                for i, variant in enumerate(variants)
+            }
+        for name, variant in variants.items():
+            variant_files = _get_files(variant.get("files", {}), challenge)
+            variant_files.update(challenge_files)
+
+            sample = Sample(
+                id=id,
+                input=variant["prompt"],
+                target=target,
+                files=variant_files,
+                metadata={
+                    "variant": name,
+                    "ratings": variant.get("ratings", {}),
+                    "tags": tags,
+                },
+                sandbox=sandbox,
+            )
+            samples.append(sample)
+
+        return samples
+
+    return [
+        sample
+        for challenge in challenges
+        for sample in create_sample_per_variant(challenge)
+    ]
+
+
+def _filter_dataset(
+    dataset: MemoryDataset,
+    filters: Callable[[Sample], bool] | list[Callable[[Sample], bool]],
+) -> MemoryDataset:
+    filters = filters if isinstance(filters, list) else [filters]
+    for filter in filters:
+        dataset = dataset.filter(filter)
+    return dataset
+
+
+def _load_challenge_info(challenge: Path) -> dict:
+    # Each challenge directory must have a config file with the challenge information.
+    for file_name in SUPPORTED_FILE_NAMES:
+        yaml_path = challenge / file_name
+        try:
+            return _load_challenge_info_from_yaml(yaml_path)
+        except FileNotFoundError:
+            continue
+    raise FileNotFoundError(
+        f"Could not find {' or '.join(SUPPORTED_FILE_NAMES)} in {challenge}."
+    )
+
+
+def _load_challenge_info_from_yaml(yaml_path: Path) -> dict:
+    with open(yaml_path, "r") as f:
+        yaml_data = f.read()
+    raw_data = yaml.safe_load(yaml_data)
+    return raw_data
+
+
+def _get_files(files_dict: dict[str, str], challenge: Path) -> dict[str, str]:
+    return {
+        key: _make_path_absolute(value, challenge) for key, value in files_dict.items()
+    }
+
+
+def _get_sandbox(challenge_info: dict, challenge: Path) -> str | tuple[str, str | None]:
+    provider, config_file = "docker", "compose.yaml"
+    # If a sandbox is provided, a provider and config_file must be specified.
+    if sandbox := challenge_info.get("sandbox"):
+        try:
+            provider, config_file = sandbox["provider"], sandbox["config_file"]
+        except KeyError:
+            raise KeyError(
+                "Could not find sandbox provider and config_file in challenge config; if a sanbox is provided, both must be specified."
```

Comment:
Consider rewording this to make it clearer.
```suggestion
                "Could not find both 'provider' and 'config_file' keys under 'sandbox' for challenge info; if a sandbox is provided, both keys must be specified."
```